pygsuite/sheets/sheet.py

```diff
@@ -1,32 +1,234 @@
+    """
+
+    if not isinstance(title, str):
+        raise TypeError("The name of the spreadsheet must be given as a string.")
+
+    request = {"properties": {"title": title}}
+
+    spreadsheet = service.spreadsheets().create(body=request, fields="spreadsheetId").execute()
+
+    id = spreadsheet.get("spreadsheetId")
+
+    return Spreadsheet(service=service, id=id)
 
 
 class Spreadsheet:
-    def __init__(self, service, id):
-        self.service = service
+    """Base class for the GSuite Spreadsheets API.
+    """
+
+    # ValueInputOption objects: https://developers.google.com/sheets/api/reference/rest/v4/ValueInputOption
+    VALUE_INPUT_OPTIONS = ["RAW", "USER_ENTERED"]
+    # Dimensions objects: https://developers.google.com/sheets/api/reference/rest/v4/Dimension
+    DIMENSIONS = ["ROWS", "COLUMNS"]
+    # ValueRenderOption objects: https://developers.google.com/sheets/api/reference/rest/v4/ValueRenderOption
+    VALUE_RENDER_OPTIONS = ["FORMATTED_VALUE", "UNFORMATTED_VALUE", "FORMULA"]
+    # DateTimeRenderOption objects: https://developers.google.com/sheets/api/reference/rest/v4/DateTimeRenderOption
+    DATE_TIME_RENDER_OPTIONS = ["SERIAL_NUMBER", "FORMATTED_STRING"]
+
+    def __init__(
+        self, id, client=None,
+    ):
+        """Method to initialize the class.
+
+        The __init__ method accepts a client connection to the Google API, which it uses to retrieve the properties
+        of the spreadsheet and create a _spreadsheet object to execute other requests of the API.
+
+        Args:
+            service (googleapiclient.discovery.Resource): connection to the Google API Sheets resource.
+            id (str): id of the Google Spreadsheet. Spreadsheet id can be found in the Spreadsheet URL between /d/ and /edit, eg:
+                https://docs.google.com/spreadsheets/d/<spreadsheetId>/edit#gid=0
+        """
+
+        from pygsuite import Clients
+
+        self.service = client or Clients.sheets_client
         self.id = id
-        self._sheet = service.open_by_key(id)
-        self._change_queue = []
 
-    def id(self):
-        return self._sheet["id"]
+        self._spreadsheet = self.service.spreadsheets().get(spreadsheetId=id).execute()
+        self._properties = self._spreadsheet.get("properties")
 
-    def flush(self):
-        out = (
+        # queues to add to and run in flush()
+        self._spreadsheets_update_queue = []
+        self._values_update_queue = []
+        # self._values_get_queue = []
+
+    def __getitem__(self, key):
+
+        # if the key is an integer, use the int as worksheet index
+        if isinstance(key, int):
+            return self.worksheets[key]
+        # if the key is a string, try to match to a worksheet name
+        elif isinstance(key, str):
+            try:
+                return [worksheet for worksheet in self.worksheets if worksheet.name == key][0]
+            except ValueError:
+                raise ValueError(
```

Comment:
I think you want this to be a keyerror for a lookup failure - more standard for catching.